presto-main/src/test/java/io/prestosql/operator/scalar/TestMathFunctions.java

```diff
@@ -1275,8 +1275,10 @@ public void testLeast()
     public void testGreatestWithNaN()
     {
         assertFunction("greatest(1.5E0, 0.0E0 / 0.0E0)", DOUBLE, Double.NaN);
+        assertFunction("greatest(1.5E0, 0.0E0 / 0.0E0, 2.7E0)", DOUBLE, Double.NaN);
         assertFunction("greatest(1.5E0, REAL '0.0' / REAL '0.0')", DOUBLE, Double.NaN);
-        assertFunction("greatest(null, REAL '0.0' / REAL '0.0')", REAL, Float.NaN);
+        assertFunction("greatest(1.5E0, REAL '0.0' / REAL '0.0', 2.7E0)", DOUBLE, Double.NaN);
+        assertFunction("greatest(null, REAL '0.0' / REAL '0.0')", REAL, null);
```

Comment:
Unrelated to this commit

Reply:
It's testing that the loop condition handling is correct when there are more than two values.